### ISSUE TITLE
🐛 Rome format doesn't work properly with ignore files

### ISSUE THREAD

**Issue by IvanKuzyshyn** [rome/tools#4132]
### Environment information

```block
CLI:
  Version:              11.0.0
  Color support:        true

Platform:
  CPU Architecture:     x86_64
  OS:                   macos

Environment:
  ROME_LOG_DIR:         unset
  NO_COLOR:             unset
  TERM:                 "xterm-256color"

Rome Configuration:
  Status:               loaded
  Formatter disabled:   false
  Linter disabled:      true

Workspace:
  Open Documents:       0
```

### What happened?

1. Rome was initialised using `rome init` command
2. Having this configuration file:
```json
{
  "$schema": "./node_modules/rome/configuration_schema.json",
  "formatter": {
    "enabled": true,
    "indentStyle": "tab",
    "lineWidth": 120,
    "ignore": [
      "node_modules/**",
      "public/**"
    ]
  },
  "linter": {
    "enabled": false
  }
}
```
3. When executing `rome format .` it checks formatting for all the files in `node_modules` directory. However, if trying to run `rome format node_modules` it prints this error `Error: no files were processed in the specified paths`.

It seems that when executing command like `rome format .` it doesn't filter out files path that matches paths from `ignore` property. Also, I've tried different version of path pattern matching, like `node_modules`, `./node_modules/*`, etc.

### Expected result

When executing `rome format .` it should ignore all the files path that matches any path in `formatter.ignore`.

### Code of Conduct

- [X] I agree to follow Rome's Code of Conduct

**Comment by ematipico** [comment]
Do you experience the same issue using `rome format ./`?

**Comment by IvanKuzyshyn** [comment]
@ematipico, yes I do.

**Comment by ematipico** [comment]
Would it be possible to share a repo where we can reproduce the issue? 

This will help triaging the issue

**Comment by SuperchupuDev** [comment]
I had the same issue before, apparently `directory/**` doesn't work as a path for me, having the same problems as this issue describes, changing it to just `directory` works for me

**Comment by IvanKuzyshyn** [comment]
@SuperchupuDev I've tried to use `"ignore": ["node_modules"]` and I have absolutely the same issue.

@ematipico, it is impossible to share that particular example because it's in the private repository but it's just a new Next.js application and nothing else. I can create another public one or maybe configure it on CodeSandbox and see whether I can reproduce it. Will it help you?

**Comment by ematipico** [comment]
> @ematipico, it is impossible to share that particular example because it's in the private repository but it's just a new Next.js application and nothing else. I can create another public one or maybe configure it on CodeSandbox and see whether I can reproduce it. Will it help you?

A fresh new repository with minimal code to reproduce the issue would be excellent! I am asking because I tried to create a small repository and I can't reproduce it. 

Take in consideration that, by default, `rome` ignored the `node_modules` during the traversal.

**Comment by IvanKuzyshyn** [comment]
@ematipico you know what - I was able to fix it.
I started reproducing the issue and realized that it is not reproducible in a new repo with all the same configuration, so I ended up reinstalling `rome` following these two commands:

```bash
pnpm remove --save-dev rome
pnpm add --save-dev rome
```
And now everything is working fine. In addition, I removed the `ignore` field and it still continues working as expected.

Still don't know what was the root cause but it's working for me now. Appreciate your help.
